lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -115,7 +115,7 @@ protected function createRequest($httpMethod, $path, $body = null, array $header
     public function getLastResponse($force = false)
     {
         $lastResponse =  parent::getLastResponse();
-        if (304 != $lastResponse->getStatusCode()) {
+        if ($lastResponse && 304 != $lastResponse->getStatusCode()) {
```

Comment:
I saw that these changes are for the 1.7 branch. 
I believe that this line only will fix the issue addressed. 

@spidgorny Can you provide a stacktrace of the exception?